builder/builder.go

```diff
@@ -609,3 +609,104 @@ func (b *Builder) BuildDeltaPacksPreviousVersions(prev, to uint32, printReport b
 	}
 	return nil
 }
+
+// BuildDeltaManifests between two versions of the mix.
+func (b *Builder) BuildDeltaManifests(from, to uint32) error {
+	var err error
+
+	if to == 0 {
+		to = b.MixVerUint32
+	} else {
+		if to > b.MixVerUint32 {
+			return errors.Errorf("--to version must be at most the latest mix version (%d)", b.MixVerUint32)
+		}
+	}
+	if from == to {
+		fmt.Println("the --from version matches the --to version, nothing to do")
+		return nil
+	} else if from > to {
+		return errors.Errorf("the --from version must be smaller than the --to version")
+	}
+
+	outputDir := filepath.Join(b.Config.Builder.ServerStateDir, "www")
+
+	toManifest, err := swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(to), "Manifest.MoM"))
+	if err != nil {
+		return errors.Wrapf(err, "couldn't find manifest of target version")
+	}
+
+	fromManifest, err := swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(from), "Manifest.MoM"))
+	if err != nil {
+		return errors.Wrapf(err, "couldn't find manifest of from version")
+	}
+
+	fmt.Printf("Using %d workers\n", b.NumDeltaWorkers)
+	_, err = swupd.CreateManifestDeltas(b.Config.Builder.ServerStateDir, fromManifest, toManifest, b.NumDeltaWorkers)
+	if err != nil {
+		log.Printf("%s\n", err)
+	}
+
+	return nil
+}
+
+// BuildDeltaManifestsPreviousVersions builds manifests to version from up to
+// prev versions. It walks the Manifest "previous" field to find those from versions.
+func (b *Builder) BuildDeltaManifestsPreviousVersions(prev, to uint32) error {
+	var err error
+
+	if to == 0 {
+		to = b.MixVerUint32
+	} else {
+		if to > b.MixVerUint32 {
+			return errors.Errorf("--to version must be at most the latest mix version (%d)", b.MixVerUint32)
+		}
+	}
+
+	outputDir := filepath.Join(b.Config.Builder.ServerStateDir, "www")
+	toManifest, err := swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(to), "Manifest.MoM"))
+	if err != nil {
+		return errors.Wrapf(err, "couldn't find manifest of target version")
+	}
+
+	var previousManifests []*swupd.Manifest
+	cur := toManifest.Header.Previous
+	for i := uint32(0); i < prev; i++ {
+		if cur == 0 {
+			break
+		}
+		var m *swupd.Manifest
+		m, err = swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(cur), "Manifest.MoM"))
+		if err != nil {
+			log.Printf("Warning: Could not find manifest for previous version %d, skipping...\n", cur)
+			continue
+		}
+		// do not create delta-packs over format bumps since clients can't update
+		// past the boundary anyways. Only check for inequality, if the format
+		// goes down that should be checked elsewhere.
+		if m.Header.Format != toManifest.Header.Format {
```

Comment:
shouldn't this check also exist in `BuildDeltaManifests`?

Reply:
Ah indeed, also fixing the comment.

Reply:
Hrm so I basically took the implementation from BuildDeltaPacks which doesn't do this check. Should it be added there too? (I'll probably not do that though via this PR).

Reply:
I can't think of a situation where we would need deltas across formats so it might make sense there as well.